node/network/bridge/src/validator_discovery.rs

```diff
@@ -274,15 +274,28 @@ impl<N: Network, AD: AuthorityDiscovery> Service<N, AD> {
 		}
 
 		// ask the network to connect to these nodes and not disconnect
-		// from them until removed from the priority group
-		if let Err(e) = network_service.add_to_priority_group(
-			PRIORITY_GROUP.to_owned(),
+		// from them until removed from the set
+		if let Err(e) = network_service.add_peers_set_reserved(
+			super::COLLATION_PROTOCOL_NAME.into(),
```

Comment:
Here we use the same set for collation peer set and validation peer set, I assume this is not intentional. In the initial design we passed the peer set id in the request (would be a parameter in this `on_request` function), we can bring it back.
But I still don't understand how validators are supposed to have a collators peer set without knowing their addresses. IIUC a request with `COLLATION_PROTOCOL_NAME` means that we are collators trying to connect to validators. so that would create a peer set on the collator side, not validator side.

Reply:
> I still don't understand how validators are supposed to have a collators peer set without knowing their addresses

If by "address" you mean a multiaddr (IP address/port), it's contained in the second parameter of `add_peers_set_reserved`.

Reply:
> > I still don't understand how validators are supposed to have a collators peer set without knowing their addresses
> 
> If by "address" you mean a multiaddr (IP address/port), it's contained in the second parameter of `add_peers_set_reserved`.

I thought for some reason these multiaddresses are of validators participating in the collation protocol, not collators. Is that wrong? Do we run authority discovery on all collators?

Reply:
Indeed, these are the addresses of the validators, not of the collators. The code is correct when run from the collator's point of view. From the validator's point of view, we rely on inbound slots to accept incoming connections from collators.

The code should eventually change so that in the future validators don't connect to other validators through this protocol, but only through collators.